ray-curator/ray_curator/stages/text/modules/score_filter.py

```diff
@@ -162,18 +233,20 @@ def process(self, batch: DocumentBatch) -> DocumentBatch | None:
 
         """
         df = batch.to_pandas()
-        bool_mask = self.compute_filter_mask(df)
-        result_df = df[bool_mask]
 
-        if len(result_df) == 0:
-            print(f"All documents filtered out for batch {batch.task_id}")
+        for filter_fn_i, filter_field_i, invert_i in zip(self.filter_fn, self.filter_field, self.invert, strict=True):
+            bool_mask = self.compute_filter_mask(df, filter_fn_i, filter_field_i, invert_i)
+            df = df[bool_mask]
+
+        if len(df) == 0:
+            logger.info(f"All documents filtered out for batch {batch.task_id}")
```

Comment:
Like, if its None, does our downstream pipeline logic handle it or will it fail ? 

Like, should we create a empty data object here ?